CONTRIBUTING.rst

```diff
@@ -106,12 +106,65 @@ Before you submit a pull request, check that it meets these guidelines:
    https://travis-ci.com/pyfar/pyfar/pull_requests
    and make sure that the tests pass for all supported Python versions.
 
-Tips
-----
 
-To run a subset of tests::
+Testing Guidelines
+-----------------------
+Test-Driven-Development (TDD) is the fundamental technique followed in developing the pyfar software packages to ensure a good software development practice. In principle, it is based on `three steps <https://martinfowler.com/bliki/TestDrivenDevelopment.html>`_
+
+- Write a test for the next bit of functionality you want to add.
+- Write the functional code until the test passes.
+- Refactor both new and old code to make it well structured.
+
+In the following, you'll find a more specific guideline. Note: these instructions are not generally applicable outside of pyfar.
+
+- The main tool used for testing is `pytest <https://docs.pytest.org/en/stable/index.html>`_.
+- All tests are located in the *tests/* folder.
+- Avoid dependencies on other pyfar functionalities. This allows easier debugging in case of failing tests due to errorneous implementations. The recommended workflow is given in the following sections.
```

Comment:
Trying to make the first point clearer and add two more points to the list

```suggestion
- Avoid dependencies on other pyfar functionalities. Otherwise, your test might fail in the future due to bugs in other parts of pyfar.
- Use fixtures in your tests whenever possible to avoid duplicate code (see below).
- Make sure that all important parts of pyfar are covered by the tests. This can be checked using *coverage* (see below).

The recommended workflow is given in the following sections.
```